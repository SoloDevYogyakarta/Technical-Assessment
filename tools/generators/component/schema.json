{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "component",
  "type": "object",
  "properties": {
    "filename": {
      "type": "string",
      "description": "Write your file name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your filename"
      },
      "$default": {
        "$source": "filename"
      }
    },
    "directoryname": {
      "type": "string",
      "description": "Write your directory name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your directoryname"
      },
      "$default": {
        "$source": "directoryname"
      }
    },
    "projectname": {
      "type": "string",
      "description": "Write your project name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your projectname"
      },
      "$default": {
        "$source": "projectname"
      }
    }
  }
}
