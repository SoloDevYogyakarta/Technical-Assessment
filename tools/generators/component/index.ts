import {
  Tree,
  formatFiles,
  getProjects,
  names,
  generateFiles,
} from '@nrwl/devkit';
import { join } from 'path';
import { ISchema } from '../schema';

export default async function (tree: Tree, schema: ISchema) {
  const workspace = getProjects(tree).get(schema.projectname);
  const templatePath = 'tools/generators/component/templates';
  const directoryPath = schema.directoryname
    ? `${workspace?.sourceRoot}/lib/${schema.directoryname}`
    : `${workspace?.sourceRoot}/lib`;
  const filename = `${schema.filename}.component`;
  const targetPath = join(directoryPath, filename);
  const componentNames = names(filename);
  const substitutions = {
    tmpl: '',
    ...componentNames,
  };
  generateFiles(tree, templatePath, targetPath, substitutions);
  await formatFiles(tree);
}
