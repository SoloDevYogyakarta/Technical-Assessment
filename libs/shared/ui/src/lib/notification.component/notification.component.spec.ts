import { NotificationComponent } from './notification.component';
import {
  TestBed,
  TestBedStatic,
  ComponentFixture,
} from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { NgHeroiconsModule } from '@dimaslz/ng-heroicons';

describe('NotificationComponent', () => {
  let container: TestBedStatic;
  let component: ComponentFixture<NotificationComponent>;

  beforeEach(async () => {
    container = await TestBed.configureTestingModule({
      imports: [NgHeroiconsModule],
      declarations: [NotificationComponent],
    }).compileComponents();

    component = TestBed.createComponent(NotificationComponent);
    component.detectChanges();
  });

  it('should render snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it("should have as title 'NotificationComponent'", () => {
    const template = component.componentInstance;
    template.open = 1;
    template.title = 'Success';
    template.message = 'Created notification';
    template.status = true;
    component.detectChanges();
    expect(template.title).toEqual('Success');
  });

  it('should render notification', () => {
    const fixture = component.componentInstance;
    fixture.open = 1;
    fixture.title = 'Success';
    fixture.message = 'Created notification';
    fixture.status = true;
    component.detectChanges();
    const fixtures = component.debugElement.query(
      By.css('#notification-title-testid')
    );
    const template = fixtures.nativeElement as HTMLElement;
    expect(template.textContent).toContain('Success');
  });

  afterEach(async () => {
    await jest.clearAllMocks();
  });
});
