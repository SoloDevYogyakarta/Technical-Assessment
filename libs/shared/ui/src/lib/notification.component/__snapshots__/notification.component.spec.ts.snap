// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationComponent should render snapshot 1`] = `
<notification-component
  __ngContext__="0"
  message=""
  notification={[Function NotificationService]}
  open="0"
  status="false"
  title=""
>
  <div
    class="w-full max-w-sm bg-white fixed bottom-1 left-1 rounded-lg p-2 shadow-md hidden flex-none"
    id="notification-testid"
  >
    <div
      class="w-full bg-red-600 rounded-md p-2 relative"
    >
      <div
        class="absolute top-1 right-1"
      >
        <x-mark-solid-icon
          class="cursor-pointer"
          color="white"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            style="width: 22px; height: 22px; color: white;"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.47 5.47a.75.75 0 0 1 1.06 0L12 10.94l5.47-5.47a.75.75 0 1 1 1.06 1.06L13.06 12l5.47 5.47a.75.75 0 1 1-1.06 1.06L12 13.06l-5.47 5.47a.75.75 0 0 1-1.06-1.06L10.94 12 5.47 6.53a.75.75 0 0 1 0-1.06z"
              fill-rule="evenodd"
            />
          </svg>
        </x-mark-solid-icon>
      </div>
      <label
        class="text-white font-semibold"
        for="title"
        id="notification-title-testid"
      >
      </label>
      <p
        class="text-white text-sm"
        id="notification-message-testid"
      >
      </p>
    </div>
  </div>
</notification-component>
`;
